feat: Hibernate with Panache ranged query (#7510)
quarkusio/quarkus: docs/src/main/asciidoc/hibernate-orm-panache.adoc

The `PanacheQuery` type has many other methods to deal with paging and returning streams.

=== Using a range instead of pages

`PanacheQuery` also allow range based query instead of page based.

[source,java]
----
// create a query for all living persons
PanacheQuery<Person> livingPersons = Person.find("status", Status.Alive);

> // make it use a range or start index 0 and last index 25.

Reviewer: ```suggestion
// make it use a range: start at index 0 until index 25 (inclusive).
```

livingPersons.range(0, 25);

// get the range
List<Person> firstRange = livingPersons.list();

> // to get the next range, you need to call range again

Reviewer: Oh hold on, I thought the range was inclusive on the last index? If it is, then you get element 25 for the previous range and this one too, no?

Author: Yes, the example should be `range(0, 24)` then `range(25, 49)` to have two range of 25 items as range starts at 0.
I'll update the documentation.

We can decide to make it start at 1 if needed, but in IT stuff starts at 0 ;)

Reviewer: OK fine.

List<Person> secondRange = livingPersons.range(25, 50).list();
----

[WARNING]
----

> You cannot mix range and page, if you use a range, all page related methods will throw an `UnsupportedOperationException`;

Reviewer: ```suggestion
You cannot mix ranges and pages: if you use a range, all methods that depend on having a current page will throw an `UnsupportedOperationException`;
```

> you must use a method that defines a new page before using them : `page(Page)` or `page(int, int)`.

Reviewer: ```suggestion
you can switch back to paging using `page(Page)` or `page(int, int)`.
```
